mldsa/fips202/fips202.c

```diff
@@ -403,7 +422,32 @@ static unsigned int keccak_absorb(uint64_t s[25], unsigned int pos,
     pos = 0;
   }
 
+  while (inlen >= r)
+  __loop__(
+    assigns(i, in, inlen, memory_slice(s, sizeof(uint64_t) *  MLD_KECCAK_LANES))
+    invariant(inlen <= loop_entry(inlen))
+    invariant(in == loop_entry(in) + (loop_entry(inlen) - inlen))
+  )
+  {
+    for (i = 0; i < r; i++)
+    __loop__(
+      assigns(i, in, memory_slice(s, sizeof(uint64_t) *  MLD_KECCAK_LANES))
+      invariant(i <= r)
+      invariant(in == loop_entry(in) + i)
+    )
+    {
+      s[i / 8] ^= (uint64_t)*in++ << 8 * (i % 8);
+    }
+    inlen -= r;
+    KeccakF1600_StatePermute(s);
+  }
+
   for (i = pos; i < pos + inlen; i++)
+  __loop__(
+          assigns(i, in, memory_slice(s, sizeof(uint64_t) * MLD_KECCAK_LANES))
```

Comment:
Nit: Spacing

Reply:
@manastasova I took the liberty to adjust this and similar instances in a follow-up commit.